src/corelibs/U2View/src/ov_sequence/ADVGraphModel.h

```diff
@@ -36,214 +36,214 @@ namespace U2 {
+/** Cut-off mode state of graph. Contains min & max cutoffs thresholds. */
+class U2VIEW_EXPORT GSequenceGraphMinMaxCutOffState {
+public:
+    virtual ~GSequenceGraphMinMaxCutOffState() = default;
 
-private:
-    U2SequenceObject *lastSeqObj;
-    QByteArray lastSeqData;
+    double min = 0;
+    double max = 0;
+    bool isEnabled = false;
 };
 
-class U2VIEW_EXPORT GSequenceGraphWindowData {
+/** A collection of static methods to work with graph values. */
+class U2VIEW_EXPORT GSequenceGraphUtils {
 public:
-    GSequenceGraphWindowData()
-        : step(0), window(0) {
-    }
-    GSequenceGraphWindowData(qint64 _step, qint64 w)
-        : step(_step), window(w) {
-    }
+    /** Returns number of steps (data points) to cover 'range' using 'window' & 'step'.*/
+    static qint64 getNumSteps(const U2Region &range, qint64 window, qint64 step);
 
-    virtual ~GSequenceGraphWindowData() {
-    }
+    /** Returns true if the value is 'UNDEFINED_GRAPH_VALUE'. */
+    static bool isUndefined(float value);
 
-    qint64 step;
-    qint64 window;
-};
+    /** Returns min of 2 values. Filters out 'UNDEFINED_GRAPH_VALUE'. */
+    static float getMinValue(float value1, float value2);
 
-class U2VIEW_EXPORT GSequenceGraphMinMaxCutOffData {
-public:
-    GSequenceGraphMinMaxCutOffData()
-        : minEdge(0.0), maxEdge(0.0), enableCuttoff(false) {
-    }
-    GSequenceGraphMinMaxCutOffData(double _min, double _max, bool _enableCuttoff)
-        : minEdge(_min), maxEdge(_max), enableCuttoff(_enableCuttoff) {
-    }
-
-    virtual ~GSequenceGraphMinMaxCutOffData() {
-    }
-
-    double minEdge;
-    double maxEdge;
-    bool enableCuttoff;
-};
+    /** Returns max of 2 values. Filters out 'UNDEFINED_GRAPH_VALUE'. */
+    static float getMaxValue(float value1, float value2);
+
+    /** Returns pair of min and max values in the given screen range. */
+    static QPair<float, float> getMinAndMaxInRange(const QSharedPointer<GSequenceGraphData> &graph, const U2Region &screenRange);
 
-struct PairVector {
-    PairVector();
-    QVector<float> firstPoints;  // max if use both
-    QVector<float> secondPoints;
-    QVector<float> cutoffPoints;
-    QVector<float> allCutoffPoints;
-    bool useIntervals;
+    /** Returns a value for the given x coordinate. In useIntervals mode the value is average between min & max values. */
+    static float getPointValue(const QSharedPointer<GSequenceGraphData> &graph, int x);
 
-    bool isEmpty() const;
+    /**
+     * A constant with undefined value for a point in a graph.
+     * Such points must now be drawn at all or have a tooltip.
+     * */
+    static const float UNDEFINED_GRAPH_VALUE;
 };
 
+/** Map of colors by data series name. */
 typedef QMap<QString, QColor> ColorMap;
 
-class U2VIEW_EXPORT GSequenceGraphDrawer : public QObject {
+/** Stateful graph drawing controller. Holds current setting of the graph widget. */
+class U2VIEW_EXPORT GSequenceGraphDrawer : public QObject, private GSequenceGraphUtils {
```

Comment:
`GSequenceGraphUtils` contains `static` functions only, I don't get the idea of the inheritance.

Reply:
It is a pattern to avoid adding a long class name  prefix before each utility method